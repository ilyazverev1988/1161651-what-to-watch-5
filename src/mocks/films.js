export default [
  {
    preview: `img/aviator.jpg`,
    nameFilm: `Aviator`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Thrillers`,
    releaseYear: 2014,
    descriptionFilm: `Friendship, metamorphosis, and adventure.`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Everything we do is connected with zen: sex, bliss, moonlight, sorrow.`,
    producer: `Heu, orexis!`,
    duration: `1h 39m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/dardjeeling-limited.jpg`,
    nameFilm: `Dardjeeling limited`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2016,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/bohemian-rhapsody.jpg`,
    nameFilm: `Bohemian rhapsody`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    nameFilm: `Fantastic beasts the crimes of grindelwald`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2018,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 50m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/johnny-english.jpg`,
    nameFilm: `Johnny english`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2019,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 9.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 30m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/macbeth.jpg`,
    nameFilm: `Macbeth`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2020,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 5.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 33m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/no-country-for-old-men.jpg`,
    nameFilm: `No country for old men`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2017,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 44m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/war-of-the-worlds.jpg`,
    nameFilm: `War of the worlds`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 54m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
];
