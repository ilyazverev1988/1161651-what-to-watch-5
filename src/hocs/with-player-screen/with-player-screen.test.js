import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withPlayerScreen from "./with-player-screen";
import films from "../../mocks/films";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withPlayerScreen(MockComponent);
const match = {params: {id: `4`}};

it(`withPlayerScreen is rendered correctly`, () => {
  const tree = renderer.create(
      <MockComponentWrapped
        films={films}
        match={match}
      >
        <React.Fragment/>
      </MockComponentWrapped>, {
        createNodeMock: () => {
          return {
            play: () => {},
            pause: () => {},
            src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
          };
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
