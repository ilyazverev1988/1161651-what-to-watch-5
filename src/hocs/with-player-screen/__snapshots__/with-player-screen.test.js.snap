// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPlayerScreen is rendered correctly 1`] = `
<div>
  <video
    className="player__video"
    onPause={[Function]}
    onPlay={[Function]}
    onTimeUpdate={[Function]}
    poster="img/the-grand-budapest-hotel-poster.jpg"
  />
</div>
`;
